File: test/com/peeekay/aoc2023/java/Day02Test.java
package com.peeekay.aoc2023.java;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import static org.junit.jupiter.api.Assertions.assertEquals;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class Day02Test {
    final AOCPuzzle day = new Day02(true);

    @Test
    void part1() {
        day.solve();
        assertEquals(8, day.part1());
    }

    @Test
    void part2() {
        day.solve();
        assertEquals(2286, day.part2());
    }

}

File: aoc2024/kotlin/src/test/kotlin/Day02Test.kt
package com.peeekay.aoc2024.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("Day 02")
internal class Day02Test {

    @Test
    fun `part1 test data solve gives correct answer`() {
        val day = Day02(isTest = true)
        val answer = day.partOne()

        assertEquals(2, answer)
    }

    @Test
    fun `part2 test data solve gives correct answer`() {
        val day = Day02(isTest = true)
        val answer = day.partTwo()

        assertEquals(0, answer)
    }

}
